core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -257,29 +261,27 @@ public void onMembersChange(Collection<Address> members) {
    /**
     * Removes the collector associated with the command.
     *
-    * @param id the {@link CommandInvocationId}.
+    * @param id the id from {@link CommandInvocationId#getId()}.
     */
-   public void dispose(CommandInvocationId id) {
+   public void dispose(long id) {
       if (trace) {
          log.tracef("[Collector#%s] Dispose collector.", id);
       }
       collectorMap.remove(id);
    }
 
-   private TimeoutException createTimeoutException(CommandInvocationId id) {
-      return log.timeoutWaitingForAcks(Util.prettyPrintTime(timeoutNanoSeconds, TimeUnit.NANOSECONDS), id);
-   }
-
    private abstract class Collector<T> implements Callable<Void>, BiConsumer<T, Throwable> {
 
-      final CommandInvocationId id;
+      final Long id;
       final CompletableFuture<T> future;
       final int topologyId;
+      private final long timeoutNanoSeconds;
```

Comment:
This would remove 8 bytes from each collector, you should be more excited about it ;)